library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -478,48 +420,96 @@ impl<T, A: Allocator> VecDeque<T, A> {
         });
     }
 
+    /// Writes all values from `iter` to `dst`, wrapping
+    /// at the end of the buffer and returns the number
+    /// of written values.
+    ///
+    /// # Safety
+    ///
+    /// Assumes that `iter` yields at most `len` items.
+    /// Assumes capacity is sufficient.
+    unsafe fn write_iter_wrapping(
+        &mut self,
+        dst: usize,
+        mut iter: impl Iterator<Item = T>,
```

Comment:
idea: could this be
```suggestion
        mut iter: impl iter::TrustedLen<Item = T>,
```
since it's a private method and it's probably impossible to prove the safety condition without the iterator being `TrustedLen`?